src/components/List/List.jsx

```diff
@@ -64,6 +74,27 @@ const List = () => {
 
   const clearFilter = () => setInputValue('');
 
+  const handleDelete = (id) => {
+    const clear = () => {
```

Comment:
The timeout is automatically cleared after it has fully executed, so calling clearTimeout at the end of the timeout won't actually prevent a memory leak. What we want to do is be sure that if the component unmounts before the 3 seconds have passed, that we call clearTimeout. We can do this by saving the timeout ID as a ref and then calling clearTimeout in the cleanup phase of useEffect.
This would look something like:
```javascript
const timeoutId = useRef();

useEffect(() => {
return () => {
     if(timeoutId.current) clearTimeout(timeoutId.current);
}
}, [])
const handleDelete = (id) => {
    if (window.confirm('are you sure to delete the item?')) {
      setLoad('Removing element...');
      remove(token, id)
        .then(() => {
          setLoad('');
          setSuccess('Document successfully deleted!');
        })
        .catch((err) => setError('Error removing document: ', err));
    }
    timeoutId.current = setTimeout(() => {
      setSuccess('');
      setError('');
    }, 3000);
  };

```

Reply:
It has been solved, I had never used the useRef hook, I think it is a good way to control the elements of the DOM, making a reference to the element we want to modify.

Reply:
Refs are very useful, but can also be dangerous, so be careful with them. They're useful for assigning values as static properties without causing a re-rerender like what would happen if we were to save the id as state. (what we're doing here. they can also be useful or things like focusing on elements. Refs are handy in certain cases, but generally try to avoid using them to control an element via its parent as in React we prefer actions to emanate from the child element. There's a saying in React: "Actions up, data down." I believe that React has some warning about refs in their documentation. I'll try to find that a little later for you.